### PR TITLE
Sign In UI + General Modal Updates

### PR DESCRIPTION
**Pull request by chamod-gamage**
**WORK DONE:**

- Created Sign In Modal as per Figma (all required screens)
- Updated Sign Up Modal to include all screens from Figma (added error styling)
- Updated Modal.scss to have more reusable styling for modals
- Connected Sign Up and Sign In via "(Don't) Have an Account?" at the bottom of each modal

**TO DO:**
- Redirect to Forgot Password via "Forgot Your Password" (@meganniu or @bonnie-chin  will need the link when you guys make that modal)
- Firebase service creation and connection

**Sign In**

[*Design*](https://www.figma.com/file/dcSbJOfK3Jg1gCpWOd6hAK/TPC-Employee-Experience%3A-Phase-1?node-id=6%3A149)
image

*Implementation*:
image

**Modal Input Field Errors**

[*Design*](https://www.figma.com/file/dcSbJOfK3Jg1gCpWOd6hAK/TPC-Employee-Experience%3A-Phase-1?node-id=6%3A149)
image

*Implementation*
image

### CODE REVIEW

File: client/src/pages/Modal.scss [hunk 1]
```diff
@@ -4,23 +4,66 @@
...
+  .row.bordered {
+    display: flex;
+    margin: 0px 40px;
+    flex-direction: row;
+    justify-content: space-between;
+    border: 1px solid #e9e9e9;
+    border-radius: 4px;
+  }
+
+  // .text {
```
Comment (meganniu) [inline client/src/pages/Modal.scss]: Remove commented styling if they're not needed anymore :)))))

File: client/src/App.tsx [hunk 2]
```diff
@@ -2,34 +2,56 @@ import "./App.scss";
...
-        <Route path='/signup' strict component={SignUpModal}></Route>
-        <Route path='/verify-email' component={AuthTestComponent}></Route>
-        <Route path='/password-reset-email-sent' strict component={PasswordResetEmailSentModal}></Route>
-        <Route path='/test'>
-          <RequestGroupDonorView requestGroupId="603d9b41eb57fc06447b8a23"/>
-        </Route>
-        <Route path='/'><DonorHomepage /></Route>
-      </Switch>
-    </Router>
+        <Switch>
```
Comment (meganniu) [inline client/src/App.tsx:18]: would it be possible to minimize this diff? LOL my eyes are really bad, it's hard for me to read it
No reformatting, just add the route that you need.

File: client/src/pages/Modal.scss [hunk 3]
```diff
@@ -34,28 +77,49 @@
...
     font-weight: normal;
     font-size: 16px;
-    line-height: 20px;  
+    line-height: 20px;
     margin-bottom: 40px;
   }
 
   .link {
-    color: #004D57;
+    color: #004d57;
```
Comment (meganniu) [inline client/src/pages/Modal.scss]: Could we assign `#adadad`,`#004d57`, `#7d7d7d`, and `#e9e9e9` to global scss variables in `_config.scss`?

File: client/src/pages/SignUpModal.tsx [hunk 4]
```diff
@@ -1,151 +1,196 @@
...
-            }
-            setRequirements(copy);
-        });
+  const onChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+  };
 
-        setPassword(password);
-    }
+  const requirementToTestMap = new Map([
```
Comment (meganniu) [inline client/src/pages/SignUpModal.tsx:43]: Would it be possible to put file reformatting in another PR? Lol it's hard to understand these diffs

**Review by meganniu — CHANGES_REQUESTED** [review]
Wow, thanks for the quick work @chamod-gamage!!

There are some files with big diffs that are hard to read. Could we put reformatting in a different PR?

File: client/src/App.tsx (hunk 2 above)
Comment (chamod-gamage) [reply client/src/App.tsx:18]: Mb mb lol Prettier auto does it for me

Comment (meganniu) [reply client/src/App.tsx:18]: LOL nwnw happens to me too

File: client/src/pages/SignUpModal.tsx [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
-import { createNewAccount } from '../services/auth';
-import { Redirect } from 'react-router-dom';
-import {useState} from 'react';
+import "./Modal.scss";
+import { OverlayTrigger, Popover } from "react-bootstrap";
+import React, { FunctionComponent } from "react";
+import CommonModal from "../components/organisms/Modal";
+import { createNewAccount } from "../services/auth";
+import { Redirect } from "react-router-dom";
+import { useState } from "react";
```
Comment (chamod-gamage) [inline client/src/pages/SignUpModal.tsx]: No changes here other than formatting

File: client/src/pages/SignUpModal.tsx (hunk 4 above)
Comment (chamod-gamage) [reply client/src/pages/SignUpModal.tsx:43]: this one was pretty tough lol but i think the diffs are more broken up and digestible now

File: client/src/pages/SignUpModal.tsx [hunk 6]
```diff
@@ -18,19 +18,28 @@ const SignUpModal: FunctionComponent = () => {
...
-        createNewAccount(email, password);
-        setRedirectToHome(true);
-    };
+  const handleClick = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    createNewAccount(email, password)
+      .then((result) => {
+        setErrors(result);
+        if (result.email === "" && result.password === "") {
+          setRedirect("/"); //TODO: change this to admin view endpoint
```
Comment (meganniu) [inline client/src/pages/SignUpModal.tsx]: Path for admin homepage is going to be `/admin`

File: client/src/pages/SignUpModal.tsx [hunk 7]
```diff
@@ -100,52 +109,70 @@ const SignUpModal: FunctionComponent = () => {
...
-                                name="password"
-                                className="input-field"
-                                placeholder="Enter your password"
-                                value={password}
-                                onChange={onChangePass}           
-                            />
-                        </OverlayTrigger>
-                    </div>
+              </div>
+              <div style={{ height: 40 }} />
```
Comment (meganniu) [inline client/src/pages/SignUpModal.tsx]: No inline styling please :)))

File: client/src/pages/SignInModal.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            </div>
+            <div
+              className="text redirect right"
+              onClick={() => {
+                console.log("TODO: setRedirect('/password reset link')");
+              }}
+            >
+              Forgot your password?
+            </div>
+            <div style={{ height: 15 }} />
```
Comment (meganniu) [inline client/src/pages/SignInModal.tsx]: Could we put styling in `SignInModal.scss`?

**Review by meganniu — APPROVED** [review]
Just some minor things, but otherwise LGTM!!!

thanks for all your work 💖

File: client/src/App.tsx [hunk 9]
```diff
@@ -28,8 +29,8 @@ function App(): JSX.Element {
           <RequestGroupDonorView requestGroupId="603d9b41eb57fc06447b8a23"/>
         </Route>
         <Route path='/'><DonorHomepage /></Route>
-      </Switch>
-    </Router>
+        </Switch>
+      </Router>
```
Comment (jlight99) [inline client/src/App.tsx:33]: nit: i don't think we should be making this spacing change, as the opening and closing switch and router tags no longer align

File: client/src/pages/SignInModal.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
+import "./Modal.scss";
+import React, { FunctionComponent } from "react";
+import CommonModal from "../components/organisms/Modal";
+// import { createNewAccount } from "../services/auth";
```
Comment (jlight99) [inline client/src/pages/SignInModal.tsx:4]: nit: remove unused import

File: client/src/pages/SignInModal.tsx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    <CommonModal
+      title={modalTitle}
+      subtitle={subtitle}
+      show={show}
+      handleClose={handleClose}
+      body={
+        <div>
+          <form onSubmit={handleClick}>
+            <div>
+              <div className="text signup">Email address {}</div>
```
Comment (jlight99) [inline client/src/pages/SignInModal.tsx]: nit: should be "Email address" (remove {})

File: client/src/pages/SignUpModal.tsx (hunk 5 above)
Comment (jlight99) [inline client/src/pages/SignUpModal.tsx]: nit: the `useState` import can go in the same import as React
ex. `import React, { FunctionComponent, useState } from "react";`

**Review by jlight99 — APPROVED** [review]
wow thanks for cleaning up the modal UI! just a few nitpicks from my end, everything looks really good!

File: client/src/pages/SignInModal.tsx (hunk 8 above)
Comment (chamod-gamage) [reply client/src/pages/SignInModal.tsx]: I'm thinking of just making an atom called `Spacer` that takes in height as a prop and returns the corresponding div with that height - a trick from my first co-op lol

**Comment by chamod-gamage** [comment]
Hey @meganniu @jlight99 thanks for the review! I'm going to clean up the nits and then with my final commit I'll reactivate Prettier so that the code has nice formatting again :)

File: client/src/pages/SignInModal.tsx (hunk 10 above)
Comment (chamod-gamage) [reply client/src/pages/SignInModal.tsx:4]: Thanks for the catch - but will keep this in there for now to remember to replace the import with the SignIn firebase service

File: client/src/pages/SignUpModal.tsx (hunk 7 above)
Comment (chamod-gamage) [reply client/src/pages/SignUpModal.tsx]: I'm thinking of just making an atom called Spacer that takes in height as a prop and returns the corresponding div with that height - a trick from my first co-op lol

Comment (meganniu) [reply client/src/pages/SignUpModal.tsx]: lol i don't think i've heard of that trick yet
Would it be possible to just change the margins on the surrounding elements instead of adding a spacer?

Comment (chamod-gamage) [reply client/src/pages/SignUpModal.tsx]: What I'm thinking though is that we want our components and component styling to be reusable. If we want to include an input field or button in another component, we'd want it to have minimal margin/padding attached to it as possible because adding spacing like this is a lot easier than having to end up doing workarounds in another component for margins/padding that are only there because it was convenient for one use of the component or having to do unique styling for every modal for atoms and molecules that are the same. What do you think?

File: client/src/pages/SignUpModal.tsx (hunk 4 above)
Comment (bonnie-chin) [reply client/src/pages/SignUpModal.tsx:43]: OMG SORRY Y'ALL this is so late holy here's the styling feedback
- I think the modal header font size should be bigger? is it 24px?
- are we doing the icon symbols :O ? I noticed that the alert icon wasn't there
- left margin of the filler text in the sign in is a bit too big I think? It should be the same as the margin in "Modal Input Field Errors"

**Comment by bonnie-chin** [comment]
OMG SORRY Y'ALL this is so late holy here's the styling feedback

- I think the modal header font size should be bigger? is it 24px?
- left margin of the filler text in the sign in is a bit too big I think? It should be the same as the margin in "Modal Input Field Errors"
- are we doing the icon symbols :O ? I noticed that the alert icon wasn't there
^^ styling feedback is in order of priority
